src/cargo/ops/cargo_check.rs

```diff
@@ -0,0 +1,89 @@
+use core::Workspace;
+use ops::{self, CompileOptions, MessageFormat};
+use util::important_paths::{find_root_manifest_for_wd};
+use util::{CliResult, Config};
+
+#[derive(RustcDecodable)]
+pub struct Options {
```

Comment:
I was trying to keep calls via the API as close to calls via the CLI. I'll take another look and see what I can figure out